pkg/jobs/progress.go

```diff
@@ -143,6 +146,7 @@ func (p *ProgressUpdateBatcher) Add(ctx context.Context, delta float32) error {
 		p.reported = p.completed
 		p.lastReported = timeutil.Now()
 	}
+	p.Unlock()
 
 	if shouldReport {
 		return p.Report(ctx, p.completed)
```

Comment:
The read on `p.completed` here needs to be within the mutex to prevent a race condition. Probably like `completed := p.completed` and then `p.Report(ctx, completed)` so Report doesn't lock the mutex.

Reply:
whoops, yep, done.